testing/testnode.go

```diff
@@ -0,0 +1,94 @@
+package testing
+
+import (
+	"encoding/hex"
+	"net/http"
+	"path/filepath"
+	"time"
+
+	"go.sia.tech/renterd/internal/consensus"
+	"go.sia.tech/renterd/node"
+	"lukechampine.com/frand"
+)
+
+type noopHandler struct{}
+
+func (h *noopHandler) ServeHTTP(http.ResponseWriter, *http.Request) {
+}
+
+type TestCluster struct {
+	Autopilot *node.Node
+	Bus       *node.Node
+	Workers   []*node.Node
+
+	Cleanup []func() error
+}
+
+func randomPassword() string {
+	return hex.EncodeToString(frand.Bytes(32))
+}
+
+func newTestCluster(dir string) (*TestCluster, error) {
+	// Use shared wallet key.
+	wk := consensus.GeneratePrivateKey()
+
+	// Prepare individual dirs.
+	busDir := filepath.Join(dir, "bus")
+	workerDir := filepath.Join(dir, "worker")
+	autopilotDir := filepath.Join(dir, "autopilot")
+
+	// Generate API passwords.
+	busPassword := randomPassword()
+	workerPassword := randomPassword()
+	autopilotPassword := randomPassword()
+
+	// Create bus.
+	bus := node.NewNode("127.0.0.1:0", busPassword, busDir, &noopHandler{}, wk)
+	err := bus.CreateBus(true, "127.0.0.1:0")
+	if err != nil {
+		return nil, err
+	}
+
+	// Create worker.
+	worker := node.NewNode("127.0.0.1:0", workerPassword, workerDir, &noopHandler{}, wk)
+	err = worker.AddBus(bus.APIAddress(), busPassword)
+	if err != nil {
+		return nil, err
+	}
+	if err := worker.CreateWorker(); err != nil {
+		return nil, err
+	}
+
+	// Create autopilot.
+	autopilot := node.NewNode("127.0.0.1:0", autopilotPassword, autopilotDir, &noopHandler{}, wk)
```

Comment:
Yeah so this could be `nil`. It just seemed nicer to only specify the key once in `NewNode` and then give it to whatever component needs it than have both `CreateWorker` and `CreateBus` pass it in and handle various edge cases. e.g. if the bus is not created but added then it doesn't need the wallet key but the worker still does and vice versa.